### PR TITLE
Modifiled event loop

### PR DESCRIPTION
**Pull request by tuhinmallick**
Refs #1718 
It fixes the warnings that are caused due to python 3.11

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: gtk/src/toga_gtk/dialogs.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class BaseDialog:
     def __init__(self):
-        loop = asyncio.get_event_loop()
+        loop = asyncio.new_event_loop()
```
Comment (freakboy3742) [inline gtk/src/toga_gtk/dialogs.py:9]: This one was correct - at the point you're opening a dialog, there *must* be an event loop; if an event loop *doesn't* exist, the warning/error raised by asyncio is correct. 

```suggestion
        loop = asyncio.get_event_loop()
```

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Looks good; one suggestion inline. 

The CI failure is due to Towncrier. We require every PR to include a file describing the change that has been made; this allows us to compile release notes. See [the contribution guide](https://toga.readthedocs.io/en/latest/how-to/contribute-code.html#add-change-information-for-release-notes) for details.

One last note - ordinarily, I'd be happy to make these changes myself, but in this case I can't, because you've submitted your PR from your main branch. Generally speaking, it's a bad idea to submit PRs from your main branch; you should create a fork, and submit the PR from there. This will make it much easier to maintain multiple PRs, and will also make it easier to merge any upstream changes into your own repository in the future.

File: gtk/src/toga_gtk/dialogs.py (hunk 1 above)
Comment (tuhinmallick) [reply gtk/src/toga_gtk/dialogs.py]: I made the changes

**Comment by tuhinmallick** [comment]
@freakboy3742 I have forked the repo and pushed the changes. What i did wrong is that I did not make a bug fix branch for it and pushed it to the main branch of my forked repo. Thanks for pointing it out to me and I will make sure to follow your advice.

File: changes/1725.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+It fixes the warnings that are caused due to python 3.11 for asyncio
```
Comment (freakboy3742) [inline changes/1725.bugfix.rst:1]: This release note should be in the form of a potential release note - e.g.:
```suggestion
Usage of asycnio in GTK has been updated to match Python 3.11 APIs.
```

**Review by freakboy3742 — CHANGES_REQUESTED** [review]

**Review by freakboy3742 — APPROVED** [review]
Awesome - thanks for the fix!

**Pull request merged**